ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -1062,53 +1063,110 @@ namespace OrdinalAxis {
+                const index =
+                Composition.findIndexOf &&
+                Composition.findIndexOf(ordinalPositions, val, true);
+                if (ordinalPositions[index] === val) {
+                    return index;
+                }
+                const percent =
+                    (val - ordinalPositions[index]) /
+                    (ordinalPositions[index + 1] - ordinalPositions[index]);
+                return index + percent;
+            }
             let axis = this,
                 ordinal = axis.ordinal,
+                slope = ordinal.slope,
                 ordinalPositions = ordinal.positions,
-                ret;
+                extendedOrdinalPositions = ordinal.extendedOrdinalPositions;
 
             if (!ordinalPositions) {
-                ret = val;
+                return val;
+            }
 
+            let ordinalLength = ordinalPositions.length,
+                ordinalIndex;
+            // If the searched value is inside visible plotArea, ivastigate the
+            // value basing on ordinalPositions.
+            if (
+                ordinalPositions[0] <= val &&
+                ordinalPositions[ordinalLength - 1] >= val
+            ) {
+                ordinalIndex = getIndexInArray(ordinalPositions, val);
+                // final return value is based on ordinalIndex
             } else {
 
-                let ordinalLength = ordinalPositions.length,
-                    i,
-                    distance,
-                    ordinalIndex;
+                if (!extendedOrdinalPositions) {
+                    extendedOrdinalPositions =
+                        ordinal.getExtendedPositions &&
+                        ordinal.getExtendedPositions();
+                    ordinal.extendedOrdinalPositions = extendedOrdinalPositions;
+                }
+                if (!(extendedOrdinalPositions && extendedOrdinalPositions.length)) {
 
-                // first look for an exact match in the ordinalpositions array
-                i = ordinalLength;
-                while (i--) {
-                    if (ordinalPositions[i] === val) {
-                        ordinalIndex = i;
-                        break;
-                    }
+                    return val;
                 }
 
-                // if that failed, find the intermediate position between the
-                // two nearest values
-                i = ordinalLength - 1;
-                while (i--) {
-                    if (val > ordinalPositions[i] || i === 0) { // interpolate
-                        // something between 0 and 1
-                        distance = (val - ordinalPositions[i]) /
-                            (ordinalPositions[i + 1] - ordinalPositions[i]);
-                        ordinalIndex = i + distance;
-                        break;
+                const EOPlength = extendedOrdinalPositions.length;
```

Comment:
Please use some other name to adhere to the style of using lower case first on JS const and let's. Suggest `positionsLength` or just `length`.